dead.py

```diff
@@ -55,12 +57,19 @@ def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
 
     def visit_ClassDef(self, node: ast.ClassDef) -> None:
         if not self.is_test:
-            self.defines[node.name].add(self.filename)
+            self.mark_definition(node)
         self.generic_visit(node)
 
+    def mark_definition(self, node: DefinitionAST) -> None:
```

Comment:
I think you can just use `ast.AST` as the type

Reply:
Not every `ast.AST` has a `name` attribute. For example we use `id` in `definition_str`

Reply:
then I would `def mark_definition(self, name: str, node: ast.AST) -> None:` -- as it is this is trying to do too much I think

Reply:
Meaning call it inside `visit_Call` as well and check the types inside? 
I don't understand what just changing the annotation would do

Reply:
You don't need to check the types, and you don't need the gross union

Reply:
Are you talking about `mark_definition` or `definition_string`? because if `mark_definition` is annotated with `node:ast.AST` mypy (rightfully) raises `dead.py:64: error: "AST" has no attribute "name"`

Reply:
All of them, this is over-DRYing as written

Reply:
fair enough

Reply:
and yeah the union on `def_str` was pretty gross, and completely unnecessary at that 😄 
In my defense I wrote `mark_def` first but then the `visit_attr` line was too long so instead of replacing I ended up adding quite clumsily 🛩